scipio/src/free_list.rs

```diff
@@ -0,0 +1,109 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the
+// MIT/Apache-2.0 License, at your convenience
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2020 Datadog, Inc.
+//
+use std::mem;
+use std::ops;
+
+#[derive(Debug, Copy, Clone, PartialOrd, PartialEq)]
+pub(crate) struct Idx(pub(crate) usize);
```

Comment:
This is because the type is Copy. It would be possible to make it a non-copy resource, but

* we'll need some kind of separate Copy version anyway -- conceptually, `Source` and the correspoinding `UringDescriptor` jointly own the id
* we'll have to cast it to u64 at some point anyway, so we'd also need RAII escape hatches. 

So, I don't think this'll help with correctness in this case. 

Maaaybe if we resuffle surround code enough, we make making this type linear more worthwhile, but that can be changed separately.